lib/Core/BuildEngine.cpp

```diff
@@ -1420,7 +1425,12 @@ class BuildEngineImpl {
     }
 
     auto dependencyID = getKeyID(key);
-    taskInfo->discoveredDependencies.push_back(dependencyID);
+    // Sort files which are less likely to be authored at the end, in order to find cycles based on authored files first.
+    if (ends_with(key, "module.modulemap")) {
```

Comment:
StringRef should have a method like this on it that you can just use instead (with basically no performance penalty).